### DISCUSSION TITLE
Kubernetes compatibility

### DISCUSSION THREAD

**Discussion by dAnjou** [nolar/kopf#1061] [Q&A]
Hi,

how does Kopf handle compatibility with Kubernetes' API? Couldn't find anything like a version matrix.

Thanks 🙏

**Comment by nolar** [comment]
Generally, all currently supported versions of Kubernetes are tested in CI and must work. 

Older versions might work, since the API is generally stable, but not guaranteed. 

You can use K8s’s version scew or the endoflife website:

* https://kubernetes.io/releases/version-skew-policy/
* https://endoflife.date/kubernetes


The same for Python versions.

(Except that I forgot to add K8s 1.28 and python 3.12 recently.)

**Reply by nolar** [reply]
PS: My mistake — K8s 1.28 is tested under the dynamic name "latest". However, K8s 1.27 dropped out when it stopped being the latest. I've re-added it in #1065.

Python 3.12 & PyPy 3.10 are added in #1067 (blocked) & #1066 respectively.

**Reply by dAnjou** [reply]
Ah, so basically this is the compatibility matrix: https://github.com/nolar/kopf/blob/b35ab78b9208017b8b262e997e7afb68adccaaa1/.github/workflows/thorough.yaml#L130

Since this is a bit of a concern for people, could you maybe add a note about compatibility in the docs somewhere?
